app/actions/navigation.js

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Navigation} from 'react-native-navigation';
+
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+export function resetToChannel() {
```

Comment:
change this to be non-redux dependent, this will work for now, but will need changes with multi-server support data layer

Reply:
Do you have an example of how these are going to look with the new data store? We're going to need to bind these to the new store somehow.

Reply:
it will still use thunk and you can getState will return an instance of the db, but why do we need access to the store in the navigation and why do they need to be redux actions?

Reply:
I'm not sure I understand. How would we use thunk and getState, but not redux actions?

And I'm undecided about exactly how to handle these actions, but I'm maybe 1/5 towards these all actions through connect since:
1. It makes it trivial to mock them for testing. We could use Jest to mock them if they're imported into the component directly, but that's more complicated than passing a mocked function in directly.
2. It keeps them consistent with how other actions are called. I like the idea that you can tell what side effects a component has specifically by looking at what actions are passed in.